zstackctl/zstackctl/ctl.py

```
@@ -177,7 +177,22 @@
+# Alinux4: disable foreign_key_checks to workaround SNS initialization bug
+# CentOS 8 has foreign_key_checks=0 by default, Alinux4 enables it by default,
+# causing SNS HTTPTopicAndEndPointCreator to fail on persist() due to missing platform record.
+# Note: foreign_key_checks is a session-level variable, cannot be set in my.cnf directly.
+# Use init-file to execute SET GLOBAL at MariaDB startup.
+if grep -qi 'Alibaba Cloud Linux' /etc/os-release 2>/dev/null; then
+    grep 'foreign_key_checks' $mysql_conf >/dev/null 2>&1
+    if [ $? -ne 0 ]; then
+        echo "foreign_key_checks=0 via init-file (Alinux4 workaround)"
+        echo "SET GLOBAL foreign_key_checks=0;" > /var/lib/mysql/init_fk.sql
+        chown mysql:mysql /var/lib/mysql/init_fk.sql
+        sed -i '/\[mysqld\]/a init-file=/var/lib/mysql/init_fk.sql\' $mysql_conf
```

Ruff (0.15.11)

[warning] Invalid escape sequence: `\[`

Add backslash to escape sequence

(W605)